concent_api/api-integration-force-get-task-result-test.py

```diff
@@ -135,65 +150,201 @@ def main():
+    time.sleep(CONCENT_MESSAGING_TIME)  # TaskToCompute.deadline + FORCE_ACCEPTANCE_TIME + CONCENT_MESSAGING_TIME < current time <= TaskToCompute.deadline + FORCE_ACCEPTANCE_TIME + 2 * CONCENT_MESSAGING_TIME.
+    api_request(  # Concent   -> Requestor:  ForceGetTaskResultFailed (provider does not submit anything)
         cluster_url,
         'receive',
         REQUESTOR_PRIVATE_KEY,
         CONCENT_PUBLIC_KEY,
         headers = {
-            'Content-Type':                 'application/octet-stream',
-            'concent-client-public-key':    b64encode(REQUESTOR_PUBLIC_KEY).decode('ascii')
-        }
+            'Content-Type': 'application/octet-stream',
+            'concent-client-public-key': b64encode(REQUESTOR_PUBLIC_KEY).decode('ascii')
+        },
+        expected_status  = 200,
+        expected_message = ForceGetTaskResultFailed
     )
 
 
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("cluster_url")
+    parser.add_argument("tc_patterns", nargs='*')
+    args = parser.parse_args()
+    return args.cluster_url, args.tc_patterns
+
+
+def get_tests_list(patterns, all_objects):
+    def _is_a_test(x):
+        return "case_" in x
+
+    tests = list(filter(lambda x: _is_a_test(x), all_objects))
+    if len(patterns) > 0:
+        safe_patterns = [pattern for pattern in patterns if _is_a_test(pattern)]
+        tests = [test for pattern in safe_patterns for test in tests if pattern in test]
+    return sorted(tests)
+
+
+def execute_tests(tests_to_execute, **kwargs):
+    objects = globals()
+    tests = [objects[name] for name in tests_to_execute]
+    for test in tests:
+        task_id = kwargs['task_id'] + test.__name__
+        kw = {k: v for k, v in kwargs.items() if k != 'task_id'}
+        test(task_id=task_id, **kw)
+        print("-" * 80)
+
+
+def main():
+    global CONCENT_MESSAGING_TIME
+    global FORCE_ACCEPTANCE_TIME
+    global TOKEN_EXPIRATION_TIME
+    global WAIT_TIME_1
+
+    cluster_url, patterns   = parse_arguments()
+    current_time            = get_current_utc_timestamp()
+    task_id                 = str(random.randrange(1, 100000))
+    tests_to_execute        = get_tests_list(patterns, list(globals().keys()))
+    print("Tests to be executed: \n * " + "\n * ".join(tests_to_execute))
+    print()
+    number_of_tests         = len(tests_to_execute)
+
+    cluster_consts         = get_protocol_constants(cluster_url)
+    print_protocol_constants(cluster_consts)
+    CONCENT_MESSAGING_TIME = cluster_consts.concent_messaging_time
+    FORCE_ACCEPTANCE_TIME  = cluster_consts.force_acceptance_time
+    TOKEN_EXPIRATION_TIME  = cluster_consts.token_expiration_time
+    WAIT_TIME_1            = DEADLINE_OFFSET + FORCE_ACCEPTANCE_TIME  # recalculated as it could change due to cluster_consts
+
+    execute_tests(tests_to_execute, cluster_url=cluster_url, current_time=current_time, task_id=task_id)
+
+    total_fails = count_fails.get_fails()
+    if total_fails > 0:
```

Comment:
How about moving this into the class, and just call `count_fails.print_fails()` ?

Reply:
@rwrzesien  when the fails should be printed? How could one know when all tests have been executed?

Reply:
@dybi What I have meant here is to move 

```
if total_fails > 0:
    print(f'Total failed tests : {total_fails} out of {number_of_tests}')
..

into some method inside `count_fails` class, and instead of doing this check + print in test code, do it inside method of `count_fails` class, and in test code only call this method. In example (pseudo code):

```
class count_fails(object):

    def print_total_fails(self):
         if self.get_total_fails() > 0:
              print(f'Total failed tests : {total_fails} out of {number_of_tests}')

......

count_fails.print_total_fails()
```

Reply:
done